test/src/SpendPermissions/useRecurringAllowance.t.sol

```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import {Test, console2} from "forge-std/Test.sol";
+
+import {SpendPermissions} from "../../../src/SpendPermissions.sol";
+
+import {SpendPermissionsBase} from "../../base/SpendPermissionsBase.sol";
+
+contract UseRecurringAllowanceTest is Test, SpendPermissionsBase {
+    function setUp() public {
+        _initializeSpendPermissions();
+    }
+    function test_useRecurringAllowance_success_noValue() public {}
+    function test_useRecurringAllowance_revert_unauthorizedRecurringAllowance()
+        public
+    {}
+    function test_useRecurringAllowance_revert_qithdrawValueOverflow() public {}
+    function test_useRecurringAllowance_revert_exceededRecurringAllowance()
+        public
+    {}
+    // TODO what are the various success logical cases?
+    function test_userRecurringAllowance_success() public {}
```

Comment:
Main logical branch here worth confirming is ability to work on both ETH and ERC20s.

We also want tests to describe the effects in success case
- `test_useRecurringAllowance_success_updatesState`
- `test_useRecurringAllowance_success_emitsEvent`

Reply:
since `_useRecurringAllowance` doesn't itself make the `_transferFrom`  call (and thus succeeds trivially regardless of the value for `token` in the recurring allowance struct), I'm going to test the ETH v.s. token as part of the `withdraw` test paths